portfolio/src/main/webapp/scripts/featured_works.js

```diff
@@ -0,0 +1,145 @@
+    .then(response => response.json())
+    .then(projects_json => setupWorkSection(projects_json))
+    .catch(err => console.error(err));
+
+/** 
+ * Sets up featured works section by creating project elements, and
+ * left- and right- buttons for navigation.
+ */
+function setupWorkSection(projects_json) {
+  const projectsHtml = setupProjects(projects_json);
+  const worksHtml = addNavButtons(projectsHtml);
+
+  const projectsPlaceholder = document.getElementById('projects');
+  ensureNonNull(projectsPlaceholder);
+  projectsPlaceholder.innerHTML = worksHtml;
+
+  displayFirstProj();
+  setupProjectNav();
+}
+
+/** Creates string which is used to display all projects. */
+function setupProjects(projects_json) {
+  const projects = toProjects(projects_json);
+  const projectDivs = projects.map(project => project.div);
+  const projectsHtml = projectDivs.map(projectDiv => projectDiv.outerHTML)
+                          .join('');
+  return projectsHtml;
+}
+
+/** 
+ * Converts an array of projects that is represented as JSON objects to Projects. 
+ * @return {Array<Project>}
+ */
+const toProjects = (projects_json) => {
+  const projects = projects_json.map(project_json => toProject(project_json));
+  return projects;
+}
+
+/** 
+ * Converts a JSON-represented project to Project. 
+ * @return {Project}
+ */
+const toProject = (project_json) => {
+  const name = project_json.name;
+  const image = project_json.image;
+  const description = project_json.description;
+  const tools = project_json.tools;
+  const date = project_json.date;
+  const page = project_json.page;
+  return new Project(name, image, description, tools, date, page);
+}
+
+/**
+ * Adds navigation buttons.
+ * @return {string} This string is the complete HTML content required for the
+ *    featured works section.
+ */
+function addNavButtons(projectsHtml) {
+  const leftButton = '<div id="left-button" class="button">' + 
+      '<i class="fas fa-chevron-circle-left fa-lg"></i></div>';
+  const rightButton = '<div id="right-button" class="button">' + 
+      '<i class="fas fa-chevron-circle-right fa-lg"></i></div>';
+  
+  return leftButton + projectsHtml + rightButton;
+}
+
+const displayFirstProj = () => {
+  const projects = document.getElementsByClassName('project');
+  ensureNonNull(projects);
+  projects[0].style.display = 'flex';
+}
+
+/** 
+ * Sets up the navigation of projects in the featured works section using
+ * the left and right buttons.
+ */
+function setupProjectNav() {
+  const leftButton = document.getElementById('left-button');
+  ensureNonNull(leftButton);
+  leftButton.onclick = () => moveToProj(-1);
+
+  const rightButton = document.getElementById('right-button');
+  ensureNonNull(rightButton);
+  rightButton.onclick = () => moveToProj(1);
+
+  const projects = document.getElementsByClassName('project');
+  ensureNonNull(projects);
+  /** @type {number} Indicates the index of the project that is currently shown. */
```

Comment:
For comments within functions, please use // only

Reply:
Oh okay. I didn't know that. The above formatting was copied from [https://google.github.io/styleguide/jsguide.html#formatting-line-wrapping](url). Will take note of it though! (:

Reply:
Oh sorry I didn't notice that this is a property comment, if that's the case, this is totally fine. For other cases like a short explanation on what the following block of code does, // would be enough.